Library/ViewModels/LiveStreamContainerViewModel.swift

```diff
@@ -248,20 +231,72 @@ LiveStreamContainerViewModelInputs, LiveStreamContainerViewModelOutputs {
     self.creatorAvatarLiveDotImageViewHidden = hideWhenReplay
     self.numberWatchingBadgeViewHidden = hideWhenReplay
     self.availableForLabelHidden = hideWhenLive
+
+    let numberOfMinutesWatched = liveStreamControllerState
+      .filter { state in
+        switch state {
+        case .live(playbackState: .playing, _):   return true
+        case .replay(playbackState: .playing, _): return true
+        default:                                  return false
+        }
+      }
+      .flatMap { _ in
+        timer(interval: .seconds(60), on: AppEnvironment.current.scheduler)
+      }
+      .scan(0) { accum, _ in accum + 1 }
+
+    configData
+      .takePairWhen(self.deviceOrientationDidChangeProperty.signal.skipNil())
+      .observeValues { data, orientation in
+        let (project, liveStream, _, _) = data
+
+        AppEnvironment.current.koala
+          .trackChangedLiveStreamOrientation(project: project,
+                                             liveStream: liveStream,
+                                             toOrientation: orientation)
+    }
+
+    configData
+      .map { project, liveStream, _, refTag in (project, liveStream, refTag) }
+      .takePairWhen(numberOfMinutesWatched)
+      .map { tuple, minute in (tuple.0, tuple.1, tuple.2, minute) }
+      .take(during: Lifetime(self.token))
```

Comment:
We could achieve the same behaviour using a `takeUntil(until:)` and a `viewDidDisappear` signal, however as the issue here is that this signal never completes and we want its completion (and release) to be tied to the deinit of the viewModel, the `LifeTime` token felt more explicit.